src/main/java/picocli/CommandLine.java

```diff
@@ -4799,6 +4822,22 @@ private static String format(String formatString, Object... params) {
         public String[] getVersion() throws Exception { throw new UnsupportedOperationException(); }
     }
 
+    /**
+     * Provides a way to modify how command model is built. Commands may configure a model transformer using
+     * {@link Command#modelTransformer()} annotation attribute.
+     * @since X.X */
+    public interface IModelTransformer {
+        /**
+         * Returns CommandSpec after doing transformation.
+         * @return transformed CommandSpec
+         */
+        CommandSpec transform(CommandSpec commandSpec);
+    }
+
+    private static class NoOpModelTransforemer implements IModelTransformer {
```

Comment:
Small typo: `NoOpModelTransforemer` should be `NoOpModelTransformer`.

Reply:
Ok.